docs/source/openstack/volume-migration-using-s3.md

```diff
@@ -0,0 +1,190 @@
+|           |   public: https://stack.nerc.mghpcc.org:13808/v1/AUTH_01234567890123456789012345689012 |
+|           |                                                                                        |
++-----------+----------------------------------------------------------------------------------------+
+```
+
+## Configure minio client
+
+For each environment, create a MinIO alias configuration using the base URL of the "public" interface of the object-store service and the access key and secret your generated earlier:
+
+```
+$ mc alias set kaizen https://kzn-swift.massopen.cloud 1234567890ABCDEF1234567890ABCDEF 1234567890ABCDEF1234567890ABCDEF
+$ mc alias set nerc https://stack.nerc.mghpcc.org:13808 FEDCBA0987654321FEDCBA0987654321 FEDCBA0987654321FEDCBA0987654321
+```
+
+## Create volume backup in source environment
+
+Create a volume backup in kaizen:
+
+```
+$ openstack --os-cloud kaizen volume backup create --container <bucket_name> <volume id or name>
+```
+
+In the above command line, `<bucket_name>` is the name of the object storage container that will hold your backup. This must be unique across all users, so using your username as a prefix is probably a good idea (e.g., `--container larsks-backups`).
+
+This will create a set of objects in the named bucket. You can use the MinIO client to inspect it:
+
+```
+$ mcli ls kaizen
+[2022-03-17 13:48:47 EDT]     0B larsks-backups/
+$ mcli ls kaizen/larsks-backups/
+[2022-03-21 20:02:55 EDT]     0B volume_bb2086dc-d6a4-4346-9809-f47fbde673f3/
+```
+
+## Create target container in destination environment
+
+We need to create a bucket in the target environment to receive the backup:
+
+```
+$ mcli mb nerc/larsks-backups
+```
+
+## Copy the backup
+
+Use the MinIO client to copy the objects from the MOC environment to the NERC environment:
+
+```
+$ mcli mirror kaizen/larsks-backups/volume_bb2086dc-d6a4-4346-9809-f47fbde673f3 nerc/larsks-backups/volume_bb2086dc-d6a4-4346-9809-f47fbde673f3
+```
+
+## Copy backup record from source to destination
+
+Now that we've copied the backup *data* into the NERC environment, we need to register the backup with the NERC backup service. We do this by copying metadata from Kaizen to the NERC:
+
+```
+$ openstack --os-cloud kaizen volume backup record export -f value > record.txt
+$ openstack --os-cloud nerc volume backup record import -f value $(cat record.txt)
+```
+
+## Create target volume
+
+Create a volume in the NERC environment to receive the backup. This must be the same size or larger than the original volume:
+
+```
+$ openstack --os-cloud nerc volume create myvolume
+```
+
+If this is a volume that will be used to boot a Nova server, include the `--bootable` flag:
+
+```
+$ openstack --os-cloud nerc volume create --bootable myvolume
+```
+
+
+## Restore backup
+
+Finally, restore the backup onto the new volume:
+
+```
+$ openstack --os-cloud nerc volume backup restore mybackupid myvolume
```

Comment:
As our users will have already created instances in NERC that they want these volumes to go into they can probably skip the `volume create` but they will run into a problem when they try to restore into a volume that is attached. They will recieve the error below
```
Invalid volume: Volume to be restored to must be available (HTTP 400) (Request-ID: req-55fe6c89-ebe8-4fb4-ba96-46461f86e786)
```
I don't see a force flag on `volume backup restore`. do I need to detach the nerc volume prior to restoring the recently mirrored kaizen backup?

Reply:
If they're migrating a volume over to the NERC the assumption is that a volume *does not* already exist on the destination, so they will need to create one.

Reply:
That is different than the original steps provided but now that I am in the guts of the beast I see that you are correct. I will have to rewrite [step 2](https://nerc-project.github.io/nerc-docs/openstack/migration-from-MOC-to-NERC/Step2/) of the migration steps. They assume the volume needs to exist first. So just to be clear, the steps are:

1. Gather information (UUID's) of Kaizen projects to be moved over
2. Set up NERC networking
3. Migrate Kaizen volumes to NERC creating new volumes to fill with backups
4. Create NERC instances and bring in the volumes migrated in step 3